Winium/Winium.StoreApps.Driver/EmulatorHelpers/Deployer.cs

```diff
@@ -155,6 +155,25 @@ public void SendFiles(Dictionary<string, string> files)
             }
         }
 
+        public void InstallDependencies(List<string> dependencies )
+        {
+            if (dependencies == null || !dependencies.Any())
+            {
+                return;
+            }
+
+            foreach (var dependency in dependencies)
+            {
+                InstallDependency(dependency);
+            }
+        }
+
+        public void InstallDependency(string path)
+        {
+            var appManifest = Utils.ReadAppManifestInfoFromPackage(path);
+            Utils.InstallApplication(this.deviceInfo, appManifest, DeploymentOptions.None, path);
+        }
```

Comment:
So, we will have `private InstallApplicationPackage(string path, AppManifest appManifest), then we will have `Install(string path)`?

I think `RemoteApplication` should not depend on `AppManfiest` and be a lazy property the way it is now, as it is not clear when and how they are initialised. It would be better to make `` auto properties. And then change `Install` implementation to something like
```cs
public void Install()
{
    var appManifestInfo = Utils.ReadAppManifestInfoFromPackage(this.appPath);
    this.InstallApplicationPackage(this.appPath, appManifestInfo);
    this.installed = true;
    this.remoteApplication = this.Device.GetApplication(appManifestInfo.ProductId);
    Logger.Info("Successfully deployed using Microsoft.Phone.Tools.Deploy");
}
```

Or we can make `InstallApplicationPackage` return `IAppManifestInfo` or even `IRemoteApplication` for installed app. Then we won't have to call two separate methods with same parameter `path` and we will have one method that hides everything related to how the app is installed, including the need to read manifest.

```cs
public void Install()
{
    var appManifestInfo = this.InstallApplicationPackage(this.appPath);
    this.installed = true;
    this.remoteApplication = this.Device.GetApplication(appManifestInfo.ProductId);
    Logger.Info("Successfully deployed using Microsoft.Phone.Tools.Deploy");
}
```
or
```cs
public void Install()
{
    this.remoteApplication = this.InstallApplicationPackage(this.appPath);
    this.installed = true;
    Logger.Info("Successfully deployed using Microsoft.Phone.Tools.Deploy");
}
```

If we only install the application, then it will make no difference, but when installing a bunch of dependencies, then retrieving `IRemoteApplication` for each might be unnecessary or time consuming, but I doubt that.